packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -184,6 +194,27 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
       })
       .next(() => changedDocs);
   }
+
+  /**
+   * Removes all changes in the remote document changelog through `changeId`
+   * (inclusive).
+   */
+  removeDocumentChangesThroughChangeId(
+    transaction: PersistenceTransaction,
+    changeId: number
+  ): PersistencePromise<void> {
+    assert(
+      changeId <= this._lastProcessedDocumentChangesId,
+      'Cannot remove unprocessed changes'
+    );
```

Comment:
I'm a little surprised this assert works since only the primary will remove changes, but the primary also doesn't actually need to process these changes.  Do we process them anyway or something?  Is that wasted work?

Reply:
I tested this over the course of an hour. I think this assert might become invalid as more changes accumulate. Note that the primary tab sets `_lastProcessedDocumentChangesId` to the existing highest value at startup.